addons/l10n_cl/models/account_move.py

```diff
@@ -27,23 +26,82 @@ def _get_l10n_latam_documents_domain(self):
+                elif self.partner_id.l10n_cl_sii_taxpayer_type == '2':
+                    partner_domain += [('code', 'in', ['70', '71', '56', '61'])]
+                elif self.partner_id.l10n_cl_sii_taxpayer_type == '3':
+                    partner_domain += [('code', 'in', ['35', '38', '39', '41', '56', '61'])]
+                elif not self.partner_id.l10n_cl_sii_taxpayer_type or self.partner_id.country_id != self.env.ref(
+                        'base.cl') or self.partner_id.l10n_cl_sii_taxpayer_type == '4':
+                    partner_domain += [('code', 'in', [])]
+                document_type_obj = self.env['l10n_latam.document.type'].search(partner_domain)
+                document_type_ids = document_type_obj.ids
+            domain = [('id', 'in', document_type_ids)]
         return domain
 
-    @api.constrains('type', 'l10n_latam_document_type_id')
-    def _check_invoice_type_document_type(self):
-        super()._check_invoice_type_document_type()
-        for rec in self.filtered(lambda r: r.company_id.country_id == self.env.ref(
-                'base.cl') and r.l10n_latam_document_type_id):
+    def _check_document_types_post(self):
+        for rec in self.filtered(
+                lambda r: r.company_id.country_id == self.env.ref('base.cl') and
+                          r.journal_id.type in ['sale', 'purchase']):
             tax_payer_type = rec.partner_id.l10n_cl_sii_taxpayer_type
+            vat = rec.partner_id.vat
+            country_id = rec.partner_id.country_id
             latam_document_type_code = rec.l10n_latam_document_type_id.code
-            if not tax_payer_type and latam_document_type_code not in ['35', '38', '39', '41']:
-                raise ValidationError(_('Tax payer type is mandatory for this type of document. '
-                                        'Please set the current tax payer type of this client'))
+            if (not tax_payer_type or not vat) and (country_id == self.env.ref('base.cl') and latam_document_type_code
+                                                  and latam_document_type_code not in ['35', '38', '39', '41']):
+                raise ValidationError(_('Tax payer type and vat number are mandatory for this type of '
+                                        'document. Please set the current tax payer type of this customer'))
+            if rec.journal_id.type == 'sale' and rec.journal_id.l10n_latam_use_documents:
+                if (tax_payer_type == '4' or country_id != self.env.ref('base.cl')) and \
+                        latam_document_type_code not in ['110', '111', '112']:
+                    raise ValidationError(_(
+                        'Document types for foreign customers must be export type (codes 110, 111 or 112)'))
+            if rec.journal_id.type == 'sale' and not rec.journal_id.l10n_latam_use_documents:
+                raise ValidationError(_('In Chilean localization, you need to use documents in the sales journal'
```

Comment:
This is too restrictive.  latam_use_documents determines whether to apply document types, ...

Reply:
@ren-odoo

Reply:
It IS restrictive, and provides the way that things are done as they should be.. If we remove the restriction, the users will generate lots of erroneous documents, that will cause lots of problems with legal registrations. The have the option to leave localization unused if they not want to use documents. Please trust in what I say... I had have lots of problems supporting these kind of things. This leads extensive repairs of the database, or even begin a new database to register things properly. Even when this kind of things mean work and services for us, always leads to disappointed users.

Reply:
@Danisan One thing to be clear: I don't want to take away any restrictions if l10n_latam_use_documents is set on the journal.  Are not the sale journals already configured like this automatically?  Because you can never know someone has a case where he wants a sale journal without it.  @ren-odoo

Reply:
Yes, is configured automatically but only for the first sales journal in the company.

Reply:
? I thought we agreed on removing this?